src/phasorpy/_phasorpy.pyx

```diff
@@ -924,35 +933,72 @@ cdef (float_t, float_t) _phasor_at_harmonic(
 
 @cython.ufunc
 cdef (float_t, float_t) _phasor_multiply(
-    float_t real1,
-    float_t imag1,
+    float_t real,
+    float_t imag,
     float_t real2,
     float_t imag2,
 ) noexcept nogil:
-    """Return multiplication of two phasors."""
-    return real1 * real2 - imag1 * imag2, real1 * imag2 + imag1 * real2
+    """Return complex multiplication of two phasors."""
+    return (
+        real * real2 - imag * imag2,
+        real * imag2 + imag * real2
+    )
 
 
 @cython.ufunc
 cdef (float_t, float_t) _phasor_divide(
-    float_t real1,
-    float_t imag1,
+    float_t real,
+    float_t imag,
     float_t real2,
     float_t imag2,
 ) noexcept nogil:
-    """Return division of two phasors."""
+    """Return complex division of two phasors."""
     cdef:
-        float_t denom = real2 * real2 + imag2 * imag2
+        float_t divisor = real2 * real2 + imag2 * imag2
 
-    if isnan(denom) or denom == 0.0:
-        return <float_t> NAN, <float_t> NAN
+    if divisor != 0.0:
+        # includes isnan(divisor)
+        return (
+            (real * real2 + imag * imag2) / divisor,
+            (imag * real2 - real * imag2) / divisor
+        )
 
+    real = real * real2 + imag * imag2
+    imag = imag * real2 - real * imag2
     return (
-        (real1 * real2 + imag1 * imag2) / denom,
-        (imag1 * real2 - real1 * imag2) / denom
+        NAN if real == 0.0 else real * INFINITY,
+        NAN if imag == 0.0 else imag * INFINITY
     )
 
 
+# @cython.ufunc
+# cdef (float_t, float_t) _phasor_divide_real(
+#     float_t real,
+#     float_t imag,
+#     float_t divisor,
+# ) noexcept nogil:
+#     """Return phasor coordinates divided by real number."""
+#     if divisor != 0.0:
+#         # includes isnan(divisor)
+#         return real / divisor, imag / divisor
+#     return (
+#         NAN if real == 0.0 else real * INFINITY,
+#         NAN if imag == 0.0 else imag * INFINITY
+#     )
+
+
+# @cython.ufunc
+# cdef float_t _divide(
+#     float_t dividend,
+#     float_t divisor,
+# ) noexcept nogil:
+#     """Return dividend divided by divisor."""
+#     if divisor != 0.0:
+#         # includes isnan(divisor)
+#         return dividend / divisor
+#     return NAN if dividend == 0.0 else dividend * INFINITY
+
+
```

Comment:
Is it intended to keep this lines commented?

Reply:
These lines can be removed. I needed to check that numpy divide ufuns match the Cython code we use elsewhere.